packages/wallets/keystone/src/adapter.ts

```diff
@@ -102,14 +103,23 @@ export class KeystoneWalletAdapter extends BaseMessageSignerWalletAdapter {
         this.emit('disconnect');
     }
 
-    async signTransaction<T extends Transaction>(transaction: T): Promise<T> {
+    async signTransaction<T extends Transaction | VersionedTransaction>(transaction: T): Promise<T> {
         try {
             const keyring = this._keyring;
             const publicKey = this._publicKey?.toString();
             if (!keyring || !publicKey) throw new WalletNotConnectedError();
 
             try {
-                return (await keyring.signTransaction(publicKey, transaction)) as T;
+                if (isVersionedTransaction(transaction)) {
+                    const txHex = transaction.serialize();
+                    const signature = await keyring.signTransaction(publicKey, txHex);
+                    transaction.addSignature(this._publicKey as PublicKey, signature);
+                } else {
+                    const txHex = transaction.serializeMessage();
+                    const signature = await keyring.signTransaction(publicKey, txHex);
+                    transaction.addSignature(this._publicKey as PublicKey, Buffer.from(signature));
```

Comment:
With the above change made, we can just pass our local `publicKey` and not use `this._publicKey` or cast it.